### PR TITLE
Do not maintain `GTF_VAR_MULTIREG` before lowering

### PR DESCRIPTION
**Pull request by SingleAccretion**
There is no need for this; the flag is meant to make it easy for codegen to check whether the given local is a multi-reg one (which, recall, is precisely equivalent to "the parent exists in IR, yet is promoted independently").

[No diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=16745&view=ms.vss-build-web.run-extensions-tab) except the spurious ARM64 ones, and a small TP win.

### CODE REVIEW

**Comment by SingleAccretion** [comment]
@dotnet/jit-contrib

File: src/coreclr/jit/importer.cpp [hunk 1]
```diff
@@ -1523,18 +1523,9 @@ GenTree* Compiler::impAssignStructPtr(GenTree*             destAddr,
...
-    if (dest->OperIs(GT_LCL_VAR) &&
-        (src->IsMultiRegNode() ||
-         (src->OperIs(GT_RET_EXPR) && src->AsRetExpr()->gtInlineCandidate->AsCall()->HasMultiRegRetVal())))
+    if (dest->OperIs(GT_LCL_VAR) && src->IsMultiRegNode())
     {
-        if (lvaEnregMultiRegVars && varTypeIsStruct(dest))
-        {
-            dest->AsLclVar()->SetMultiReg();
-        }
-        if (src->OperIs(GT_CALL))
```
Comment (kunalspathak) [inline src/coreclr/jit/importer.cpp:1534]: Why we no longer need to check if `src == GT_CALL`?

File: src/coreclr/jit/hwintrinsicarm64.cpp [hunk 2]
```diff
@@ -1800,27 +1800,10 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
...
-            // The following is to exclude the fields of the local to have SSA.
-            varDsc->lvIsMultiRegRet = true;
-
-            GenTreeLclVar* lclVar = gtNewLclvNode(lclNum, varDsc->lvType);
-            lclVar->SetDoNotCSE();
-            lclVar->SetMultiReg();
-
-            retNode = lclVar;
+            op1     = gtNewSimdHWIntrinsicNode(retType, op1, intrinsic, simdBaseJitType, simdSize);
+            retNode = impAssignMultiRegTypeToVar(op1, sig->retTypeSigClass DEBUGARG(CorInfoCallConvExtension::Managed));
```
Comment (kunalspathak) [inline src/coreclr/jit/hwintrinsicarm64.cpp:1806]: `impAssignMultiRegTypeToVar()` method doesn't set `varDsc->lvIsMultiRegRet = true;`

See https://github.com/dotnet/runtime/pull/66551/commits/65c0af67b66c71989e62c5d653339d1a51982347

File: src/coreclr/jit/lower.cpp [hunk 3]
```diff
@@ -3391,13 +3391,11 @@ void Lowering::LowerStoreLocCommon(GenTreeLclVarCommon* lclStore)
     DISPTREERANGE(BlockRange(), lclStore);
     JITDUMP("\n");
 
-    GenTree*   src    = lclStore->gtGetOp1();
-    LclVarDsc* varDsc = comp->lvaGetDesc(lclStore);
-
+    GenTree*   src           = lclStore->gtGetOp1();
+    LclVarDsc* varDsc        = comp->lvaGetDesc(lclStore);
     const bool srcIsMultiReg = src->IsMultiRegNode();
-    const bool dstIsMultiReg = lclStore->IsMultiRegLclVar();
```
Comment (kunalspathak) [inline src/coreclr/jit/lower.cpp:3398]: Why we stop checking for `dstIsMultiReg` now?

**Review by kunalspathak — COMMENTED** [review]
Added some questions.

File: src/coreclr/jit/lower.cpp (hunk 3 above)
Comment (SingleAccretion) [reply src/coreclr/jit/lower.cpp:3398]: Is will always be `false`: the store we'll be seeing hasn't been lowered yet.

File: src/coreclr/jit/hwintrinsicarm64.cpp (hunk 2 above)
Comment (SingleAccretion) [reply src/coreclr/jit/hwintrinsicarm64.cpp:1806]: I see it does...?

https://github.com/dotnet/runtime/blob/8f06013212ebc2813948f78ad864ebf101047e0b/src/coreclr/jit/importer.cpp#L17728-L17743

It's even redundant, technically, `impAssignStructPtr` will set it when called from `impAssignTempGen`.

Comment (kunalspathak) [reply src/coreclr/jit/hwintrinsicarm64.cpp:1806]: ah, I was referring to the PR fork that was not doing it. Never mind.

File: src/coreclr/jit/lower.cpp (hunk 3 above)
Comment (kunalspathak) [reply src/coreclr/jit/lower.cpp:3398]: Got it.

**Review by kunalspathak — APPROVED** [review]
LGTM

**Comment by kunalspathak** [comment]
@huoyaoyuan might have to rebase some of his changes in https://github.com/dotnet/runtime/pull/66551 .

**Pull request merged**